File: gem5Sim/src/microSimProject/sensor.cc
#include "microSimProject/sensor.hh"
#include "debug/Sensor.hh"
#include <stdlib.h>

Sensor::Sensor(SensorParams *params) : 
    SimObject(params),
    event(*this),
    newTemp(params->newTemp),
    curTemp(params->curTemp)
{ 
    DPRINTF(Sensor, "Sensor created\n");    
}
Sensor*
SensorParams::create()
{
   return new Sensor(this);
}
void Sensor::startup()
{
    schedule(event, rand()%600);
}
void
Sensor::processEvent()
{
    DPRINTF(Sensor, "Current temperature is %d\n", curTemp); 
    
    DPRINTF(Sensor, "New temperature detected: %d\n", newTemp);
}

File: gem5Sim/src/microSimProject/sensor.py
from m5.params import *
from m5.SimObject import SimObject

class Sensor(SimObject):
    type = 'Sensor'
    cxx_header = "microSimProject/sensor.hh"
    
    curTemp = Param.Int("current temp of the sensor")
    newTemp = Param.Int("the new sensed temperature")
